src/Stripe.net/Services/Products/ProductService.cs

```diff
@@ -23,54 +23,56 @@ public ProductService(string apiKey)
         {
         }
 
+        public override string BasePath => "/products";
+
         public virtual Product Create(ProductCreateOptions options, RequestOptions requestOptions = null)
         {
-            return this.Post($"{Urls.BaseUrl}/products", requestOptions, options);
+            return this.CreateEntity(options, requestOptions);
         }
 
         public virtual Task<Product> CreateAsync(ProductCreateOptions options, RequestOptions requestOptions = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return this.PostAsync($"{Urls.BaseUrl}/products", requestOptions, cancellationToken, options);
+            return this.CreateEntityAsync(options, requestOptions, cancellationToken);
         }
 
         public virtual Product Delete(string productId, RequestOptions requestOptions = null)
         {
-            return this.DeleteEntity($"{Urls.BaseUrl}/products/{WebUtility.UrlEncode(productId)}", requestOptions);
+            return this.DeleteEntityNew(productId, requestOptions);
```

Comment:
Flagging that I had to remove the encoding for now. I think it makes more sense to do it at the next level. What do you think?

Reply:
`InstanceUrl` already takes care of encoding the ID! :)